common/djangoapps/student/tests/test_email.py

```diff
@@ -132,6 +137,24 @@ def _assert_activation_email(self, subject, body_fragments):
         for fragment in body_fragments:
             self.assertIn(fragment, msg.body)
 
+    @mock.patch('student.tasks.log')
+    def test_send_email_to_inactive_user(self, mock_log):
+        """
+        To verify that activation email has been sent to
+        an un-activated user(logged-in via social-auth)
```

Comment:
that's not what this test is doing, and it's not really testing the functionality you added. The logic you added sends an email if the user attempts to access the dashboard and that user is not active. You should be checking that, when you try to access the dashboard as an inactive student, the `compose_and_send_email` function is called with expected parameters.